web/src/main/ui/app/components/flows-new/validators/new-step-dialog.validator.ts

```diff
@@ -3,19 +3,24 @@ import { StepType } from '../models/step.model';
 import * as _ from "lodash";
 
 export function NewStepDialogValidator(group: FormGroup) {
-  let stepType: typeof StepType = StepType;
   let errors = {};
-  if (group.value.stepDefinitionType === stepType.MAPPING ||
-      group.value.stepDefinitionType === stepType.MASTERING) {
-    if (!group.value.targetEntity) {
-      errors['noTargetEntity'] = true;
-    }
-    if ( 
-          (group.value.selectedSource === 'collection' && !group.value.sourceCollection) ||
-          (group.value.selectedSource === 'query' && !group.value.sourceQuery)
-        ) {
-      errors['noSource'] = true;
-    }
+
+  switch (group.value.stepDefinitionType) {
+    case StepType.MAPPING:
+    case StepType.MASTERING:
+      if (!group.value.targetEntity) {
+        errors['noTargetEntity'] = true;
+      }
+      if (
+        (group.value.selectedSource === 'collection' && !group.value.sourceCollection) ||
+        (group.value.selectedSource === 'query' && !group.value.sourceQuery) ||
+        (group.value.selectedSource === '')
```

Comment:
As Aries mentioned, choosing a selected source isn't required.   But, if they do choose a source type, they must have a sourceCollection value with collection type and sourceQuery with query type.

Reply:
so the statement

> (group.value.selectedSource === '')

should be removed